packages/headless/src/controllers/triggers/headless-redirection-trigger.test.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  RedirectionTrigger,
+  buildRedirectionTrigger,
+} from './headless-redirection-trigger';
+import {buildMockSearchAppEngine, MockEngine} from '../../test/mock-engine';
+import {SearchAppState} from '../../state/search-app-state';
+import {redirection, configuration} from '../../app/reducers';
+import {logTriggerRedirect} from '../../features/redirection/redirection-analytics-actions';
+
+describe('RedirectionTrigger', () => {
+  let engine: MockEngine<SearchAppState>;
+  let redirectionTrigger: RedirectionTrigger;
+
+  function initRedirectTrigger() {
+    redirectionTrigger = buildRedirectionTrigger(engine);
+  }
+
+  function getLogTriggerRedirectAction() {
+    return engine.actions.find(
+      (a) => a.type === logTriggerRedirect.pending.type
+    );
+  }
+
+  function registeredListeners() {
+    return (engine.subscribe as jest.Mock).mock.calls.map((args) => args[0]);
+  }
+
+  beforeEach(() => {
+    engine = buildMockSearchAppEngine();
+    initRedirectTrigger();
+  });
+
+  it('initializes', () => {
+    expect(redirectionTrigger).toBeTruthy();
+  });
+
+  it('it adds the correct reducers to the engine', () => {
+    expect(engine.addReducers).toHaveBeenCalledWith({
+      redirection,
+      configuration,
+    });
+  });
+
+  it('exposes a #subscribe method', () => {
+    expect(redirectionTrigger.subscribe).toBeTruthy();
+  });
+
+  it('when the #engine.state.redirection.redirectTo is already initialized, it calls #onRedirect and dispatches #logTriggerRedirect', () => {
+    const listener = jest.fn();
+    engine.state.redirection.redirectTo = 'https://www.google.com';
+    redirectionTrigger.subscribe(listener);
+
+    expect(listener).toHaveBeenCalledTimes(1);
+    expect(getLogTriggerRedirectAction()).toBeTruthy();
```

Comment:
It would be possible to avoid logging an analytics event immediately on subscribe if [this line](https://github.com/coveo/ui-kit/pull/875/files#diff-2fb706cfa8a422f58f67ea3947be15ab9bb346e8c5cfe51f615c2ad941a16946R49) was initialized to the value in state instead of null.

Reply:
Not sure I get this, we should log and call the listener on subscribe if the redirection state did contain a trigger (not null, or equivalent would be not an empty string)

Use cases could be SSR or creating a trigger later, after the latest search